File: comandor/src/main/java/com/komandor/komandor/data/database/users/UserDao.java
package com.komandor.komandor.data.database.users;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import io.reactivex.Flowable;
import io.reactivex.Maybe;
import io.reactivex.Single;

@Dao
public interface UserDao {
  
  @Insert(onConflict = OnConflictStrategy.REPLACE)
  void insert(User user);
  
  @Query("select * from " + User.TABLE_NAME + " where user_id = :userPID")
  Flowable<User> getUserByPID(int userPID);
  
  @Query("select * from " + User.TABLE_NAME + " limit 1")
  Maybe<User> getActiveUser();
}

File: comandor/src/main/java/com/komandor/komandor/data/database/dao/UserDao.kt
package com.komandor.komandor.data.database.dao

import androidx.room.Dao
import androidx.room.Query
import androidx.room.Transaction
import com.komandor.komandor.data.database.model.User

@Dao
abstract class UserDao: BaseDao<User>() {

    @Query("select * from ACCOUNTS where pid = :userPID")
    abstract fun getUserByPID(userPID: Int): User?

    @Query("select * from ACCOUNTS where current = :current")
    abstract fun getCurrent(current:Boolean = true): User?

}